test/integration/note_test.rb

```diff
@@ -0,0 +1,39 @@
+require 'test_helper'
+
+class PregnancyNoteTest < ActionDispatch::IntegrationTest
+  before do
+    @user = create :user, email: 'testcaseworker@email.com'
+    log_in_as @user
+    @patient = create :patient, name: 'tester', primary_phone: '1231231234'
+    @pregnancy = create :pregnancy, appointment_date: nil, patient: @patient
+    visit edit_pregnancy_path(@pregnancy)
+  end
+ 
+  describe 'add patient pregnancy notes' do
+    it 'should display a pregnancy notes form and current notes' do
+        click_link 'Notes'
+        
+        within('#notes')do
+            assert_text 'Notes' #confirm notes header is visible 
+            assert has_button? 'Submit'
+        end
+    end 
+    
+    it 'should let you add a new pregnancy note' do
+        fill_in 'note_body', with: 'Sample Note Body'
+       
+        assert_difference 'Note.count', 1 do 
+          click_on 'Submit'
+        end 
+        assert_text 'Sample Note Body'
+    end 
+
+    it 'should let you edit a pregnancy note' do
+        click_on 'edit'
+        fill_in 'note_body', with: 'Sample Note Body Edit'
+        click_on 'Submit'
+        
+        assert_text 'Sample Note Body Edit'
```

Comment:
I'd consider reloading the page as well to ensure that it stays persistent between pageloads.